### DISCUSSION TITLE
Fail connecting to server

### DISCUSSION THREAD

**Discussion by LLY28** [milvus-io/milvus#18441] [Q&A and General discussion]
I occasionally fail when I call the API interface several times in a row.

In the **flask**, the Milvus API interface is invoked, and an error message is reported:

![image](https://user-images.githubusercontent.com/69830759/181451413-88ecbf7a-1cd4-4112-a39b-93eb18a87d97.png)

When an error occurs, the **docker log** looks like this:

![5POZR%Y}3$Q0GFNMH5%9JHD](https://user-images.githubusercontent.com/69830759/181449835-7378385e-44d8-4b27-835f-f980f7093190.png)

**Environment:**

![_6LVTC`8D`F9M94CV@Q0I5H](https://user-images.githubusercontent.com/69830759/181448708-eacea7c9-be09-4215-aaa7-a2c060df9319.png)

**Is CPU resource insufficient? How can I circumvent this problem.**

**Comment by LLY28** [comment]
def search(self, collection_name: str, target_vec: List[List[float]], topK: int):
        connections.connect("default", host=self.SERVER_ADDR, port=self.SERVER_PORT)
        collection_status = utility.has_collection(collection_name)
        if not collection_status:
            raise Exception(f" collection '{collection_name}' is not exist!")
        chatbot_milvus = Collection(collection_name)  # Get an existing collection.
        chatbot_milvus.load()
        search_params = {
            "metric_type": self.metic_type,
            "params": {"nprobe": self.nprobe},
        }
        search_result = chatbot_milvus.search(target_vec, "embeddings", search_params, limit=topK)[0]
        ids = search_result.ids
        distances = search_result.distances
        chatbot_milvus.release()
        connections.disconnect("default")
        return distances, ids
        
会在每次请求的时候 connect、load、release、disconnect，跟这些频繁的操作有关系吗

**Comment by yhmo** [comment]
The client time out in _wait_for_channel_ready(), it indicates the client cannot access the address 192.168.35.266:19530, or the milvus server is not ready.

Firstly check the log, if the server successfully started, there should be two lines printed in the log file:
```
[2022/08/01 10:13:14.888 +08:00] [DEBUG] [proxy/service.go:299] ["start Proxy server done"]
[2022/08/01 10:13:14.888 +08:00] [INFO] [components/proxy.go:54] ["Proxy successfully started"]
```

Your client script is no problem.

**Reply by LLY28** [reply]
请问有专门的日志文件吗，我的日志是用docker logs 命令查看的

**Reply by LLY28** [reply]
今天测试的时候还出现个奇怪的问题。分别创建一个7k+条向量和一个9w+条向量的collection后，7k+的collection可以正常search，9w+的collection调用search的时候，会卡住，一直不返回结果，debug发现一直卡在这里

search_result = chatbot_milvus.search(target_vec, "embeddings", search_params, limit=topK)[0]

入参格式校验过都正常，请问这会是什么原因

**Comment by LLY28** [comment]
这是search的代码，    

   ```
 def search(self, collection_name: str, target_vec: List[List[float]], topK: int):
        # 选择连接
        connections.connect("searcher", host=self.SERVER_ADDR, port=self.SERVER_PORT)
        connect_name = 'searcher'
        collection_status = utility.has_collection(collection_name, using=connect_name)
        print("## collection_status: ", collection_status)
        if not collection_status:
            raise Exception(f" collection '{collection_name}' is not exist!")
        chatbot_milvus = Collection(collection_name, using=connect_name)  # Get an existing collection.
        chatbot_milvus.load()
        print(f"##########Number of entities in Milvus: {chatbot_milvus.num_entities}")  # check the num_entites

        search_params = {
            "metric_type": self.metic_type,
            "params": {"nprobe": self.nprobe},
        }
        start_time = time.time()
        search_result = chatbot_milvus.search(target_vec, "embeddings", search_params, limit=topK)[0]
        print(f"only search vectors cost time: {time.time() - start_time}")
        ids = search_result.ids
        distances = search_result.distances
        chatbot_milvus.release()
        return distances, ids
```


现在猜想可能是：

- 部署milvusV2.0的时候，设置了某些参数，导致search异常
- 创建collection有问题
- search有问题

但是暂时这三个方向都还没找到问题 @yhmo

**Reply by yhmo** [reply]
@PEN170 
看起来客户端代码没什么问题，但从给的这些信息看不出是哪里出了问题。你是用的 v2.1 standalone吧？你用docker logs把search卡住时完整的日志输出来，在issues里创建一个issue，把日志贴里面，我们assign给相关人员。
